src/integration/kotlin/pl/edu/agh/gem/integration/client/FinanceAdapterClientIT.kt

```diff
@@ -0,0 +1,120 @@
+package pl.edu.agh.gem.integration.client
+
+import io.kotest.assertions.throwables.shouldThrow
+import io.kotest.matchers.shouldBe
+import org.springframework.http.HttpStatus.BAD_REQUEST
+import org.springframework.http.HttpStatus.INTERNAL_SERVER_ERROR
+import pl.edu.agh.gem.external.dto.finance.toDomain
+import pl.edu.agh.gem.helper.group.DummyGroup.GROUP_ID
+import pl.edu.agh.gem.integration.BaseIntegrationSpec
+import pl.edu.agh.gem.integration.ability.stubGetActivities
+import pl.edu.agh.gem.integration.ability.stubGetBalances
+import pl.edu.agh.gem.integration.ability.stubGetSettlements
+import pl.edu.agh.gem.internal.client.FinanceAdapterClient
+import pl.edu.agh.gem.internal.client.FinanceAdapterClientException
+import pl.edu.agh.gem.internal.client.RetryableFinanceAdapterClientException
+import pl.edu.agh.gem.util.createActivitiesResponse
+import pl.edu.agh.gem.util.createBalancesResponse
+import pl.edu.agh.gem.util.createSettlementsResponse
+
+class FinanceAdapterClientIT(
+    private val financeAdapterClient: FinanceAdapterClient,
+) : BaseIntegrationSpec({
+    should("retrieve activities successfully") {
+        // given
+        val activitiesResponse = createActivitiesResponse()
+        stubGetActivities(activitiesResponse, GROUP_ID)
+
+        // when
+        val result = financeAdapterClient.getActivities(GROUP_ID)
+
+        // then
+        result shouldBe activitiesResponse.toDomain()
+    }
+
+    should("throw FinanceAdapterClientException on client error for getActivities") {
+        // given
+        stubGetActivities(null, GROUP_ID, BAD_REQUEST)
```

Comment:
i think approach with null its more accurate because in cases with errors request we dont have "standard" response in body  . In other places changed to null